### PR TITLE
Add support for the rails 5.2 polymorphic name method

### PR DESCRIPTION
**Pull request by doliveirakn**
### What problem does this solve

Prior to Rails 5.2, there wasn't a great consensus on what to use for the polymorphic_name of a class. In some instance it was the class name, in other instances, it seemed like it was the name that was used for single table inheritance. 

As of Rails 5.2, there is now a canonical way which is to call `polymorphic_name` on the class. Now that there is a canonical way, we should remove the redundant lookups

### How does this solve it

We don't want to completely break support, so this will check if a class responds to `polymorphic_name` and if so, it'll use that. It'll keep all of the other fallbacks in place though.

### CODE REVIEW

**Comment by Drew-Goddyn** [comment]
`so this PR introduces a configuration variable that defaults to off`

Is this still the case or do we just check that it responds to polymorphic_name instead now?

**Review by Drew-Goddyn — APPROVED** [review]

**Comment by doliveirakn** [comment]
@Drew-Goddyn that was the case the first attempt at support but is no longer the case. I'll remove that from the description.

File: lib/polymorphic_integer_type/extensions.rb [hunk 1]
```diff
@@ -45,6 +46,7 @@ def belongs_to(name, scope = nil, **options)
             define_method "#{name}=" do |record|
               super(record)
               send("#{foreign_type}=", record.class)
+              association(name).loaded!
```
Comment (daniel-almeida) [inline lib/polymorphic_integer_type/extensions.rb:49]: What's the reason behind this change?

**Review by daniel-almeida — APPROVED** [review]
Left a clarifying question.

File: lib/polymorphic_integer_type/extensions.rb (hunk 1 above)
Comment (doliveirakn) [reply lib/polymorphic_integer_type/extensions.rb:49]: if we write the association in the `super(record)` and then change then `*_type` column, it will mark the association as stale, and lose the reference of it (causing it to want to lookup the object again). 

By marking the assocaition as loaded afterwards, we guarantee that if once you write the record, it maintains that association.
